system/ui/lib/list_view.py

```diff
@@ -11,154 +9,266 @@
     super().__init__()
-    self.width = width
+    self.title = title
+    self._icon = icon
+    self.description = description
+    self.show_desc = False
+
     self._enabled_source = enabled
+    self._visible_source = visible
+    self._font = gui_app.font(FontWeight.NORMAL)
+
+    # Cached properties for performance
+    self._prev_max_width: int = 0
+    self._wrapped_description: str | None = None
+    self._prev_description: str | None = None
+    self._description_height: float = 0
 
   @property
   def enabled(self):
-    return _resolve_value(self._enabled_source, False)
+    return _get_value(self._enabled_source, True)
+
+  @property
+  def is_visible(self):
+    return _get_value(self._visible_source, True)
+
+  def set_enabled(self, enabled: bool):
+    self._enabled_source = enabled
+
+  def get_desc(self):
+    return _get_value(self.description, "")
+
+  def set_icon(self, icon: str):
+    self._icon = icon
+
+  def set_desc(self, description: StrSrc):
+    self.description = description
+    current_description = self.get_desc()
+    if current_description != self._prev_description:
+      self._update_description_cache(self._prev_max_width, current_description)
 
-  def get_width(self) -> int:
-    return self.width
+  def _update_description_cache(self, max_width: int, current_description: str):
+    """Update the cached description wrapping"""
+    self._prev_max_width = max_width
+    self._prev_description = current_description
+    content_width = max_width - ITEM_PADDING * 2
+
+    # Account for icon width
+    if self._icon:
+      content_width -= ICON_SIZE + ITEM_PADDING
+
+    wrapped_lines = wrap_text(self._font, current_description, ITEM_DESC_FONT_SIZE, content_width)
+    self._wrapped_description = "\n".join(wrapped_lines)
+    self._description_height = len(wrapped_lines) * ITEM_DESC_FONT_SIZE + 10
+
+  def _get_height(self, max_width: int) -> float:
+    if not self.is_visible:
+      return 0
+
+    if not self.show_desc:
+      return ITEM_BASE_HEIGHT
+
+    current_description = self.get_desc()
+    if not current_description:
+      return ITEM_BASE_HEIGHT
+
+    if current_description != self._prev_description or max_width != self._prev_max_width:
+      self._update_description_cache(max_width, current_description)
+
+    return ITEM_BASE_HEIGHT + self._description_height - (ITEM_BASE_HEIGHT - ITEM_DESC_V_OFFSET) + ITEM_PADDING
+
+  def _render(self, rect: rl.Rectangle):
+    # Handle click on title/description area for toggling description
+    if self.description and rl.is_mouse_button_released(rl.MouseButton.MOUSE_BUTTON_LEFT):
+      mouse_pos = rl.get_mouse_position()
 
+      text_area_width = rect.width - self.get_action_width() - ITEM_PADDING
+      text_area = rl.Rectangle(rect.x, rect.y, text_area_width, rect.height)
+      if rl.check_collision_point_rec(mouse_pos, text_area):
+        self.show_desc = not self.show_desc
```

Comment:
move to `_handle_mouse_release`?

Reply:
Moving it to _handle_mouse_release isn’t ideal at the moment, but it can be addressed later. On one hand, I want to handle mouse events before rendering so that the item’s height can be recalculated. On the other hand, I think [PR #35526](https://github.com/commaai/openpilot/pull/35526) might offer a better solution. PR 35526 is able to track and detect whether mouse press and release events occur within the same rect.

Reply:
With PR https://github.com/commaai/openpilot/pull/35526, we can use `mouse.is_click_in` anywhere to detect mouse events, similar to QApp's global `QCursor`, instead of only in handle_mouse_click. This provides greater flexibility.